src/Storage/TSDatabaseView.m

```diff
@@ -183,19 +218,12 @@ + (YapDatabaseViewSorting *)messagesSorting {
           TSInteraction *message1 = (TSInteraction *)object1;
           TSInteraction *message2 = (TSInteraction *)object2;
 
-          NSDate *date1 = [self localTimeReceiveDateForInteraction:message1];
-          NSDate *date2 = [self localTimeReceiveDateForInteraction:message2];
+          uint64_t timestamp1 = message1.timestampForSorting;
+          uint64_t timestamp2 = message2.timestampForSorting;
 
-          NSComparisonResult result = [date1 compare:date2];
```

Comment:
See above!  Also, I've dried up all of this comparison code with `compareForSorting:`.